src/Avalonia.Controls.ColorPicker/ColorSpectrum/ColorSpectrum.Properties.cs

```diff
@@ -93,6 +93,14 @@ public partial class ColorSpectrum
                 nameof(Shape),
                 ColorSpectrumShape.Box);
 
+        /// <summary>
+        /// Defines the <see cref="ThirdComponent"/> property.
+        /// </summary>
+        public static readonly StyledProperty<ColorComponent> ThirdComponentProperty =
+            AvaloniaProperty.Register<ColorSpectrum, ColorComponent>(
+                nameof(ThirdComponent),
+                ColorComponent.Component3); // Value
```

Comment:
Just for case, styled property with a private setter can still be modified by the user.
Simply by calling `SetValue(ThirdComponentProperty, value)`, skipping the CLR property call.
Instead, you can make this property a direct property without a setter - it won't be possible to modify the property without hack like reflection.

Something like this:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ProgressBar.cs#L26-L30

Reply:
Yea, I considered making this a direct property but didn't want to restrict any use cases in the styles or themes. I'm not sure if it would ever be used that way but wanted to allow it. Since it is supposed to be get-only a direct property probably makes sense -- but I don't want to change the API in the future here as a breaking change. So overall decided most powerful is best. 

I also know it can be overridden by calling SetValue() directly. That was actually intentional in case some control downstream ever needs to modify the property. There is still a way to do it. So probably there should be a protected set there too.